Feature/FA-9-Occasions

Ahmed-Abdelghany-11 opened the pull request:
<!--📌 Pull Request Template -->

<!--📝 Description -->

This pull request introduces the **Occasions** feature, which includes building out the data, domain, and presentation layers to support occasions and associated product views. It also includes enhancements and cleanups across modules.

### ✅ Feature: FA-9-Occasions

- **Occasion Layer Implementation:**
  - feat(occasions): add data layer for occasions.
  - feat(occasions): add domain layer for occasions.
  - feat(occasions): implement OccasionCubit and related use cases for occasions and products.
  - feat(occasions): enhance OccasionCubit and OccasionScreen with tabbed product views.

- **Code Quality Improvements:**
  - refactor(auth): apply dart fix, format, and analyze.

---

## 📌 Pull Request Checklist

### ✅ General
- [x] This pull request is all my own work -- I have not plagiarized it.
- [x] The PR title follows the required format: `feat/FA-9-Occasions`.
- [x] The branch name follows the required format: `feature/FA-9-occasions`.

### 📂 Code Quality
- [x] All filenames are in `snake_case`.
- [x] All functions and variable names follow Dart naming conventions.
- [x] All new code is formatted with `dart format .`.

### 🛠 CI/CD Validation
- [x] Linting passes with `dart analyze`.
- [x] The branch name is validated against the allowed patterns.
- [x] The PR title is validated against the required format.

### 🚀 Testing
- [ ] Unit tests are included for all new features and bug fixes.
- [ ] `flutter test --coverage` has been run and the results reviewed.
- [ ] The project builds successfully without errors.

### 📝 Additional Information
This PR focuses on implementing the full flow for **Occasions** including product tabs, data fetching, state management using `OccasionCubit`, and related domain logic. It also includes cleanup of the authentication module via formatting and fixes for better maintainability.

### 🔗 Related JIRA Ticket(s):
- [x] JIRA Issue: `FA-9-Occasions`

### 📸 Screenshots (if applicable):
<!-- Add screenshots to show the changes visually. If not applicable, you can delete this section. -->
<p align="center">
  <img src="https://github.com/user-attachments/assets/1c1aadbd-6e91-489c-ad56-a3f71319734b" alt="login" width="300" />
</p>

Ahmed-Abdelghany-11 committed:
feat(occasions): add data layer for occasions

Ahmed-Abdelghany-11 committed:
feat(occasions): add domain layer for occasions

Ahmed-Abdelghany-11 committed:
feat(occasions): implement OccasionCubit and related use cases for occasions and products

Ahmed-Abdelghany-11 committed:
feat(occasions): enhance OccasionCubit and OccasionScreen with tabbed product views

Ahmed-Abdelghany-11 committed:
refactor(auth): apply dart fix ,format and analyze

ali72-20 commented on lib/features/occasions/data/occasion_data_source/occasion_remote_data_source.dart:
make interface for datasource and implement it to avoid violate dependacy invertion principle

ali72-20 commented on lib/features/occasions/domain/repo/occasion_repo.dart:
```diff
@@ -0,0 +1,10 @@
+import '../entity/occasions_entity.dart';
+import '../entity/products_entity.dart';
+
+abstract class OccasionRepo {
```
make interface more better if you don's have any attributes

Ahmed-Abdelghany-11 replied:
done

ali72-20 commented on lib/features/occasions/presentation/view/screen/occasion_screen.dart:
```diff
@@ -0,0 +1,68 @@
+      create: (_) => viewModel..doIntent(OccasionRequestAction()),
+      child: BlocBuilder<OccasionCubit, OccasionState>(
+        builder: (context, state) {
+          final viewModel = context.read<OccasionCubit>();
```
you already have an object from view model

ali72-20 commented on lib/features/occasions/presentation/view/widgets/product_item.dart:
```diff
@@ -0,0 +1,97 @@
+                crossAxisAlignment: CrossAxisAlignment.start,
+                children: [
+                  Text(
+                    productEntity!.title ?? "",
```
the empty Sring is hard coded insted make a variable with defualte value with "" value

ali72-20 commented on lib/features/occasions/presentation/view/widgets/product_item.dart:
```diff
@@ -0,0 +1,97 @@
+                  Row(
+                    children: [
+                      Text(
+                        'EGP ${productEntity!.priceAfterDiscount ?? ""}',
```
EGP is hard coded in arabic should be جنيه مصري

Ahmed-Abdelghany-11 replied:
done

ali72-20 commented on lib/features/occasions/presentation/view/widgets/product_view.dart:
```diff
@@ -0,0 +1,76 @@
+
+    return BlocBuilder<OccasionCubit, OccasionState>(
+      builder: (context, state) {
+        if (viewModel.products == null || state.baseState is BaseLoadingState) {
```
you need to handel null in special if condtion 
case: 
if state not laoding but products is null you need to show screen that products is empty not loading indicator

ali72-20 commented on lib/features/occasions/presentation/view/widgets/product_view.dart:
```diff
@@ -0,0 +1,76 @@
+        if (viewModel.products!.isEmpty) {
+          return Center(
+            child: Text(
+              'No products available',
```
hard coded string avoid it

ali72-20 reviewed (approved):
small changes will make it more better 
good work

ahmedelazab1220 commented on lib/features/occasions/presentation/view/widgets/product_item.dart:
```diff
@@ -0,0 +1,97 @@
+                  ),
+                ),
+              ),
+            ),
```
use cached network instead of make that

Ahmed-Abdelghany-11 committed:
feat(localization): update Arabic and English translations, add new locale keys

ahmedelazab1220 commented:
Great work ya Ahmed 👏

Ahmed-Abdelghany-11 committed:
refactor(product): adjust padding in ProductItem and optimize GridView layout

Ahmed-Abdelghany-11 committed:
feat(product): integrate cached network image for product display

Ahmed-Abdelghany-11 committed:
Merge branch 'refs/heads/developement' into feature/fa-9-occasions

# Conflicts:
#	assets/translations/ar.json
#	assets/translations/en.json
#	lib/core/utils/di/di.config.dart
#	lib/core/utils/l10n/locale_keys.g.dart
#	lib/main.dart
#	pubspec.yaml

Ahmed-Abdelghany-11 committed:
feat(occasion): Implement occasion screen UI and integrate with API

- Updated dependencies and configurations.
- Updated translations for "NoProductsAvailable".
- Added a skeleton loader for product items.
- Integrated API calls for occasions and products.
- Handled loading and error states in the UI.

Ahmed-Abdelghany-11 committed:
feat(occasion): Add custom occasion tab bar and update state management

Ahmed-Abdelghany-11 committed:
Merge branch 'developement' into feature/fa-9-occasions

# Conflicts:
#	lib/core/utils/di/di.config.dart
#	lib/core/utils/l10n/locale_keys.g.dart
#	lib/features/auth/data/data_source/remote/auth_remote_data_source_impl.dart
#	lib/features/auth/data/model/register/register_response_dto/register_response_dto.g.dart
#	lib/features/product_details/presentation/view/screens/product_details_screen.dart
#	lib/features/product_details/presentation/view/widgets/image_view.dart
#	lib/features/product_details/presentation/view/widgets/product_details_body.dart

Ahmed-Abdelghany-11 committed:
refactor(occasions): update product handling and improve data structure

Ahmed-Abdelghany-11 committed:
refactor(occasions): enhance occasion handling and improve routing logic
